### ISSUE TITLE
Mutex is not owned Exception

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/couchbase-lite-net#762]
Using 1.3.0.1-1

When running p2p sanity test

```
Unhandled Exception:
System.ApplicationException: Mutex is not owned
  at System.Threading.Mutex.ReleaseMutex () <0x1b2dd90 + 0x0005a> in <filename unknown>:0
  at (wrapper remoting-invoke-with-check) System.Threading.Mutex:ReleaseMutex ()
  at Couchbase.Lite.Listener.CouchbaseLiteResponse.WriteToStream (System.Byte[] data) <0x2f040a8 + 0x0011f> in <filename unknown>:0
  at Couchbase.Lite.Listener.CouchbaseLiteResponse.WriteData (IEnumerable`1 data, Boolean finished) <0x31e8af8 + 0x0013f> in <filename unknown>:0
  at Couchbase.Lite.Listener.DBMonitorCouchbaseResponseState.SendHeartbeatResponse (System.Object state) <0x34dc000 + 0x0009b> in <filename unknown>:0
  at System.Threading.Timer+Scheduler.TimerCB (System.Object o) <0x1b32690 + 0x00164> in <filename unknown>:0
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () <0x1970570 + 0x00035> in <filename unknown>:0
  at System.Threading.ThreadPoolWorkQueue.Dispatch () <0x196eba0 + 0x001d8> in <filename unknown>:0
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () <0x19703e0 + 0x00017> in <filename unknown>:0
```

**Comment by sethrosetter** [comment]
This does not happen with `1.3.0-148`

**Comment by sethrosetter** [comment]
Fixed in 1.3.0.1-2

**Comment by borrrden** [comment]
This still needs to be guarded against in master, so reopening until then.
